cli/internal/cmd/apply.go

```diff
@@ -214,28 +214,28 @@ func runApply(cmd *cobra.Command, _ []string) error {
 
 	upgradeID := generateUpgradeID(upgradeCmdKindApply)
 	upgradeDir := filepath.Join(constants.UpgradeDir, upgradeID)
-	clusterUpgrader, err := cloudcmd.NewClusterUpgrader(
-		cmd.Context(),
-		constants.TerraformWorkingDir,
-		upgradeDir,
-		flags.tfLogLevel,
-		fileHandler,
-	)
-	if err != nil {
-		return fmt.Errorf("setting up cluster upgrader: %w", err)
+
+	newClusterApplier := func(ctx context.Context) (clusterUpgrader, error) {
```

Comment:
nit: to make clear that this is related to the infra phase
```suggestion
	newInfrastructureApplier := func(ctx context.Context) (infrastructureApplier, error) {
```

Reply:
was there a reason to make this a factory function? I think we want to avoid this pattern, @daniel-weisse? Afaik, we only do this for the other clients because their instantiation relies on information that is only available at a later stage (e.g. kubeconfig file or some flag argument)

Reply:
thinking about it, it might be useful though if we want to avoid failures of the client instantiation (e.g. the terraform dir is in an invalid state) when we ignore the infra phase anyways.